test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_full_dispatch.cpp

```diff
@@ -0,0 +1,367 @@
+//
+// Copyright (c) 2024 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <extensionHelpers.h>
+#include "mutable_command_basic.h"
+
+#include <CL/cl.h>
+#include <CL/cl_ext.h>
+
+#include <vector>
+
+namespace {
+
+////////////////////////////////////////////////////////////////////////////////
+// command buffer with full mutable dispatch tests which handle cases:
+// CL_MUTABLE_DISPATCH_GLOBAL_OFFSET_KHR
+// CL_MUTABLE_DISPATCH_GLOBAL_SIZE_KHR
+// CL_MUTABLE_DISPATCH_LOCAL_SIZE_KHR
+// CL_MUTABLE_DISPATCH_ARGUMENTS_KHR
+// CL_MUTABLE_DISPATCH_EXEC_INFO_KHR
+
+struct MutableCommandFullDispatch : InfoMutableCommandBufferTest
+{
+    using InfoMutableCommandBufferTest::InfoMutableCommandBufferTest;
+
+    MutableCommandFullDispatch(cl_device_id device, cl_context context,
+                               cl_command_queue queue)
+        : InfoMutableCommandBufferTest(device, context, queue),
+          svm_buffers(context), group_size(0)
+    {}
+
+    bool Skip() override
+    {
+        cl_mutable_dispatch_fields_khr mutable_capabilities;
+        cl_mutable_dispatch_fields_khr requested =
+            CL_MUTABLE_DISPATCH_GLOBAL_OFFSET_KHR
+            | CL_MUTABLE_DISPATCH_GLOBAL_SIZE_KHR
+            | CL_MUTABLE_DISPATCH_LOCAL_SIZE_KHR
+            | CL_MUTABLE_DISPATCH_ARGUMENTS_KHR
+            | CL_MUTABLE_DISPATCH_EXEC_INFO_KHR;
+        bool mutable_support =
+            !clGetDeviceInfo(
+                device, CL_DEVICE_MUTABLE_DISPATCH_CAPABILITIES_KHR,
+                sizeof(mutable_capabilities), &mutable_capabilities, nullptr)
+            && (mutable_capabilities & requested) == requested;
```

Comment:
Ideally rather than mandating that all the capabilities are supported, we'd tests as many capabilities as the device supports. For example, if an implementation didn't support updating global size, global offset, and local size, then we'd only test updating arguments, exec info and svm arguments in the same update call.

Reply:
Done, please correct me if I am mistaken, with this correction there are 27 combinations of test execution. I think I went through all of the cases to confirm test passes.

Reply:
Thanks, I hadn't worked out all the combinations but it's great you've tried them all 🙂 
I want to try this test out locally before hitting approve, but I think it's looking good.